src/modm/driver/encoder/encoder.lb

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2019, Thomas Sommer
+#
+# This file is part of the modm project.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# -----------------------------------------------------------------------------
+
+def init(module):
+    module.name = ":driver:encoder"
+    module.description = """
+Software graycode ported from C-Code of Peter Dannegger:
```

Comment:
As a note: This is interpreted as Markdown and the first line should be the heading, which is displayed in `lbuild discover` and used in all docs. Something like:

```
    module.description = """
# Graycode Encoder Input

{Description} blabla, use pins on the same port, blabla, Inspired by Peter: See url.
    """
```

Reply:
What's wrong here? Not the proper doxygen outlining or what?

Reply:
It's not wrong, it's just that the modm.io module reference (and doxygen) take the first line of this description as the heading, and you don't want the first sentence to be used as the heading:
Example: https://modm.io/reference/module/modm-driver-encoder-output/ where the first line of the module description is `# Encoder Output`.

Reply:
Got this :)